packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -80,13 +80,44 @@ const initAPICallTracing = parentSpan => {
   }
 }
 
+const deferredAction = type => (...args) => {
+  emitter.emit(`ENQUEUE_NODE_MUTATION`, {
+    type,
+    payload: args,
+  })
+}
+
+const NODE_MUTATION_ACTIONS = [
+  `createNode`,
+  `deleteNode`,
+  `deleteNodes`,
+  `touchNode`,
+  `createParentChildLink`,
+  `createNodeField`,
+]
+
+const deferActions = actions => {
+  const deferred = { ...actions }
+  NODE_MUTATION_ACTIONS.forEach(action => {
+    deferred[action] = deferredAction(action)
+  })
+  return deferred
+}
```

Comment:
Is it possible to always queue these events and listen to the queue at the correct times inside the state-machine?

Reply:
Yes, always queuing them would be the ideal. The reason I didn't initially was backward-compatibility. Currently `build` doesn;t use the state machine, so would need another mechanism to watch the queue. If we agree that this is the way to go though I'm fine with havign a watcher that can be used independently for now as well.